Quaver.Shared/Screens/Select/SelectScreen.cs

```diff
@@ -536,5 +556,96 @@ public void SelectRandomMap()
                     break;
             }
         }
+
+        /// <summary>
+        ///     Starts the deleting process.
+        /// </summary>
+        private void DeleteSelected()
+        {
+            var view = View as SelectScreenView;
+            var type = view.ActiveContainer;
+
+            var selectedMapsetIndex = view.MapsetScrollContainer.SelectedMapsetIndex;
+            var selectedDifficultyIndex = view.DifficultyScrollContainer.SelectedMapIndex;
+            var selectedMapset = AvailableMapsets[selectedMapsetIndex];
+            var selectedDifficulty = selectedMapset.Maps[selectedDifficultyIndex];
+
+            var mapsetPath = Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory);
+            var difficultyPath = Path.Combine(mapsetPath, selectedMapset.Maps[selectedDifficultyIndex].Path);
+
+            // Commence deleting and reloading.
+            var confirmDelete = new ConfirmCancelDialog($"Are you sure you want to delete this {( type == SelectContainerStatus.Mapsets ? "mapset" : "difficulty" )} ? ", (sender, confirm) =>
+            {
+                var mapTitle = selectedMapset.Maps.First().Title;
+
+                // Externally loaded map check.
+                if (MapManager.Selected.Value.Game != MapGame.Quaver)
+                {
+                    // Display error message.
+                    NotificationManager.Show(NotificationLevel.Error, "This map was loaded from an external source, it cannot be deleted.");
+                    return;
+                }
+
+                // Dispose of the background for the currently selected map.
+                BackgroundHelper.Background?.Dispose();
+
+                // Dispose of the currently playing track.
+                AudioEngine.Track?.Dispose();
+
+                // Run path deletion in the background.
+                ThreadScheduler.Run(() => DeletePath(type == SelectContainerStatus.Difficulty && selectedMapset.Maps.Count > 1 ? difficultyPath : mapsetPath));
+
+                // Remove mapset/difficulty from cache and AvailableMapsets list.
+                if (type == SelectContainerStatus.Mapsets || type == SelectContainerStatus.Difficulty && selectedMapset.Maps.Count == 1)
```

Comment:
This condition occurs twice (this line and 3 lines above), which is why I suggested making it into a variable. Please do that.